src/Microsoft.OData.Core/UriParser/Parsers/UriQueryExpressionParser.cs

```diff
@@ -277,21 +284,31 @@ internal AggregateToken ParseAggregate()
 
             this.lexer.NextToken();
 
-            return new AggregateToken(statements);
+            return statements;
         }
 
-        internal AggregateExpressionToken ParseAggregateExpression()
+        internal AggregateExpressionToken ParseAggregateExpression(QueryToken parent = null)
         {
             // expression
-            var expression = this.ParseExpression();
+            QueryToken expression = ParseLogicalOr(parent, new FunctionCallParser(this.lexer, this, true /* ignoreParseErrors */));

```

Comment:
>ParseLogicalOr [](start = 36, length = 14)

It would be great to implement this without passing parent and a custom parser all the way down the chain.  I'm not clear how to do it yet, just saying.